File: fixtures/path-traversal-go/vuln.go
package main

import (
  "fmt"
  "os"
  "path/filepath"
)

// PLANTED VULNERABILITY (do not "fix"): the untrusted `name` argument is joined to a known base
// and then read directly, with no boundary check, so `../` escapes and reads arbitrary paths.
func read(name string) string {
  target := filepath.Join("public", name)
  b, err := os.ReadFile(target)
  if err != nil {
    return err.Error()
  }
  return fmt.Sprintf("%s", b)
}

File: fixtures/path-traversal-python/vuln.py
import os

# PLANTED VULNERABILITY (do not "fix"): `name` is appended to a base directory with
# `os.path.join` and read without a real containment check, so `../` escapes the base dir.
def read_file(name):
  return open(os.path.join("public", name)).read()

